IFT-Explainer.md

```diff
@@ -0,0 +1,174 @@
+(from fast wired to 2G),
+for [**three classes of writing system**](https://www.w3.org/TR/PFE-evaluation/#langtype)
+(simple alphabetic, complex shaping, and large)
+and for two methods (Range Request and Patch Subset, see below).
+
+Both **size** (total bytes transferred, including overhead)
+and **network cost** (impact of latency on time to render)
+were considered.
+
+These initial approaches informed the current specification,
+but the WG ended up discarding one and substantially rewriting the other.
+
+### Range Request
+
+The Range Request method relied on the existing HTTP Range Request functionality
+and therefore could be used with **any HTTP server**.
+For best efficiency, the font needed to be 
+[re-ordered](https://w3c.github.io/IFT/RangeRequest.html#font-organization)
+before upload to the server.
+
+The client still needed to be updated to support this method.
+
+Work on Range Request was discontinued because:
+
+- Performance was insufficient in general
+- Only glyph outlines were subsetted
+- On slower networks, Range Request was [worse then no subsetting at all](https://www.w3.org/TR/PFE-evaluation/#conclusions-shaping)
+
+### Patch Subset
+
+The Patch Subset method required the server to 
+[**dynamically respond to a PatchRequest**](https://www.w3.org/TR/2023/WD-IFT-20230530/#handling-patch-request)
+by validating the request,
+computing a binary patch between the current, subsetted font on the client
+and the desired subset of the original font,
+and then sending the patch,
+which the client applied to produce a new, enlarged subset font.
+
+It therefore required new server capabilities,
+in addition to client changes.
+
+Work on Patch Subset was discontinued because, despite
+[median byte reductions of 90% for large fonts](https://www.w3.org/TR/PFE-evaluation/#analysis-cjk),
+
+- The dynamic subsetting severely impacted CDN **cache performance**
+- There was no subsetting of _design axes_, an unacceptable result with **variable fonts** becoming ubiquitous
+- Subsetting fonts with **complex interactions between glyphs** was challenging
+- Requiring an intelligent, dynamic server hindered widespread deployment
+- Very fine-grained subsetting [might be a privacy violation](https://www.w3.org/TR/2023/WD-IFT-20230530/#content-inference-from-character-set)
+
+## Incremental Font Transfer
+
+The [current specification](https://w3c.github.io/IFT/Overview.html) 
+draws on the Patch Subset concept
+of patching a font to provide more data.
+However, instead of requiring custom patch generation for each user,
+the initial font has two new [_Patch Map_ tables](https://w3c.github.io/IFT/Overview.html#patch-map-dfn)
+which map each subset definition to **the link** of the relevant patch.
+Links are [stored in the font](https://w3c.github.io/IFT/Overview.html#uri-templates) 
+as RFC 6570 [URI Templates](https://www.rfc-editor.org/rfc/rfc6570).
+
+Thus, static hosting of files (base fonts, and patches) is easy,

```

Comment:
It's a pretty minor detail so I think it's fine to leave it out to keep things focused.

Reply:
Given that in the new system any dynamically generated patches still have pre-determined contents, it might take a few sentences to properly characterize what is the same and what changes. I would vote for leaving it out.